content/raw/wallet/add_account.md

```diff
@@ -0,0 +1,123 @@
+---
+title: "Add Account spec"
+description: ""
+date: 2020-12-08T08:48:57+00:00
+lastmod: 2020-12-08T08:48:57+00:00
+draft: false
+images: []
+menu:
+  raw:
+    parent: "wallet"
+weight: 100
+toc: true
+---
+
+# Add Account Spec
+
+## Motivation
+
+Users often want to add an account to their wallet and this is a very common use case. These accounts can either be generated, or imported. In some cases Users also want to add addresses just to watch them.
+
+## Definitions
+
+- ethereum address - hexadecimal string of 64 characters used to identify an account
+- private key - hexadecimal string of 64 characters that is used to sign transactions belonging to the account
+- seed phrase - a list of 12 words used to generate one or more private keys
+- derivation path - a list of integers used to derive a private key from a seed phrase 
+- ENS - a ethereum domain name that is used to resolve an address
+- assets - refers to Ethereum Tokens and ERC-20 Tokens
+- active account - an ethereum address that has transactions
+
+## Designs
+
+- [designs](https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?node-id=49%3A1737962)
+
+## Use Cases
+
+### Generate an account
+
+**TODO** design is missing
+
+1. User selects "Generate an account"
+2. System shows dialog to enter account name and emoji
+3. System randomly selects an emoji
+4. User fills in account name
+5. (optional): User selects emoji
+6. User selects "Add account"
+
+### Import an account from a private key
+
+1. User selects "Add with a key or seed phrase"
+2. System shows dialog to enter a private key or seed phrase, name and emoji
+3. System randomly selects an emoji
+4. User enters private key
+4.1 alternative: user fills seed phrase, see "Import an account from a seed phrase"
+5. Systems validates private key
```

Comment:
what happens when private key is not valid?

In general in this specs, there is only the happy path. When there is user input, should the non happy paths be specified?